pkg/operator/ceph/cluster/controller.go

```diff
@@ -527,23 +528,46 @@ func (c *ClusterController) checkPVPresentInCluster(drivers []string, clusterID
 	return false, nil
 }
 
-// removeFinalizer removes a finalizer
-func removeFinalizer(client client.Client, name types.NamespacedName) error {
-	cephCluster := &cephv1.CephCluster{}
-	err := client.Get(context.TODO(), name, cephCluster)
+func removeFinalizers(client client.Client, name types.NamespacedName) error {
+	// Remove cephcluster finalizer
+	err := removeFinalizer(client, name, &cephv1.CephCluster{}, "")
+	if err != nil {
+		return errors.Wrap(err, "failed to remove cephcluster finalizer")
+	}
+
+	// Remove finalizer for rook-ceph-mon secret
+	name = types.NamespacedName{Name: mon.AppName, Namespace: name.Namespace}
+	err = removeFinalizer(client, name, &corev1.Secret{}, mon.DisasterProtectionFinalizerName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to remove finalizer for the secret %q", name.Name)
+	}
+
+	// Remove finalizer for rook-ceph-mon-endpoints configmap
+	name = types.NamespacedName{Name: mon.EndpointConfigMapName, Namespace: name.Namespace}
+	err = removeFinalizer(client, name, &corev1.ConfigMap{}, mon.DisasterProtectionFinalizerName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to remove finalizer for the configmap %q", name.Name)
+	}
+	return nil
+}
+
+func removeFinalizer(client client.Client, name types.NamespacedName, obj client.Object, finalizer string) error {
+	err := client.Get(context.TODO(), name, obj)
 	if err != nil {
 		if kerrors.IsNotFound(err) {
-			logger.Debug("CephCluster resource not found. Ignoring since object must be deleted.")
+			logger.Debugf("%s resource not found. Ignoring since object must be deleted.", name.Name)
 			return nil
 		}
-		return errors.Wrapf(err, "failed to retrieve ceph cluster %q to remove finalizer", name.Name)
+		return errors.Wrapf(err, "failed to retrieve %q to remove finalizer", name.Name)
+	}
+	if finalizer == "" {
+		finalizer = opcontroller.BuildFinalizerName(obj.GetObjectKind().GroupVersionKind().Kind)
```

Comment:
How about if the CephCluster passes its finalizer name to this method just like the other resources, instead of calculating it here?

Reply:
Changed the way by calling the RemoveFinalizer method, if the string passed is empty.